### DISCUSSION TITLE
TryAsync prelude should have an overload that takes a Task

### DISCUSSION THREAD

**Discussion by pilotMike** [louthy/language-ext#1278] [Q&A] [answered]
There is a possible unexpected use case where passing a failed Task into TryAsync can be interpreted as a success:
![image](https://github.com/louthy/language-ext/assets/3967294/c2de3a3b-cb0e-4258-baab-802d1c547869)

Since there's no overload taking a Task, the Task goes in to the `'A` overload and is interpreted as a success. 
There should be something like `TryAsync<Unit> TryAsync(Task t) => ...`

**Comment by louthy** [comment, marked as answer]
I realise that usage might be confusing, but there's reasons why I'm not going to change it:

1. [I'm planning on deprecating the Async based monadic types](https://github.com/louthy/language-ext/discussions/1269).  I'm midway through this process on [another branch](https://github.com/louthy/language-ext/blob/v5-transducers), so making changes  to functionality that will be deprecated is just looking for merge headaches that I don't need.
2. Really, as functional programmers, _everything_ should be an expression and therefore _everything_ should return a value.  There is an extension to `Task` called `ToUnit()` that you should apply to any `Task` you get from a 3rd party API as soon as you hold the value - it will convert your `Task` to `Task<Unit>`.  Then make sure all the APIs under your control return values.  It's a discipline for sure.

If you still fell that you desperately need this overload: a good technique is to create your own `Prelude` and then add it a `global using static`, that will allow you to relatively seamlessly extend the capabilities of LanguageExt without having to manually haul in an extra `using` for each `cs` file.

**Reply by pilotMike** [reply]
Thank you for the reply, that's a good workaround if we need it 👍
